lib/internal/loader/Loader.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const URL = require('url').URL;
+
+const ModuleWrap = process.binding('module_wrap').ModuleWrap;
+
+const ModuleMap = require('internal/loader/ModuleMap');
+const ModuleJob = require('internal/loader/ModuleJob');
+const resolveRequestUrl = require('internal/loader/resolveRequestUrl');
+const {Error} = global;
+
+const pathToFileURL = (pathname) => `file://${pathname}/`;
+
+const getNamespaceOf = (m) => {
+  const tmp = new ModuleWrap('import * as _ from "";_;', '');
+  tmp.link(async () => m);
+  tmp.instantiate();
+  return tmp.evaluate();
+};
+
+function getBase() {
+  try {
+    return pathToFileURL(process.cwd());
+  } catch (e) {
+    // If the current working directory no longer exists.
```

Comment:
I feel like the regular modular loader may explode in this case?

We should keep the behavior consistent whatever it is.